### ISSUE TITLE
taking too much time

### ISSUE THREAD

**Issue by Ubnlqs** [aw1875/puppeteer-hcaptcha#23]
```javascript
const solveCaptcha = require('hcaptcha-solver');

(async () => {
    try {
      const response = await solveCaptcha('https://2captcha.com/');
      console.log(response);
      // F0_eyJ0eXAiOiJKV1Q...
    } catch (error) {
      console.log(error);
    }
})();
```
it takes lot of time ( 40 seconds ) 
Can I fix it?

**Comment by aw1875** [comment]
It's TensorFlow related. With Google's Cloud Vision it would finish around 5-10 seconds but Google recently changed their system so people have to play and I figured people wouldn't want to do that. If enough people are interested I can reintroduce Google's Vision again and have the option between that or TensorFlow.

**Comment by JokersCat** [comment]
Oh that would be amazing I’m completely new in js and only know Python quite good.So some code looks simular to py so yeah I will try to use it I guess and thx for the work and please add google vision back 😂

**Comment by JokersCat** [comment]
I think people are interested to pay because there are websites where you have to solve a hcaptcha to get registered and this registered accounts can get soled for a bit 😅

**Comment by aw1875** [comment]
@JokersCat Honestly, the main issue is that you can't use threading in Node. If threading was *really* possible I could definitely speed up the process by running all the image detection processes at once. I was thinking about maybe looking into trying to incorporate something like calling a C# class for this. Also, I'm almost finished finals so I will try to work on adding the option for Cloud Vision again after (or the C# class) I finish.

**Comment by JokersCat** [comment]
Oh okay I saw that you made something before in Python would it be possible to you to make maybe a Python library on that? 😅

**Comment by aw1875** [comment]
@JokersCat this project was based off of [this python repo](https://github.com/Futei/SineCaptcha) initially. I haven't worked with Python in a very long time so I probably wouldn't attempt to make this into a Python library

**Comment by JokersCat** [comment]
Oh okay 🥲 And the Python thing you linked is complete different I think it’s patched (I don’t tested it) so the thing worked like only requesting with the Site-key + Site and then hcaptcha gives you the pass 😂 but your thing is complete different with image recognition etc. so it would be Great if you could do it in Python.Anyways I will try to make it but I don’t got the time to do it 🥲 but I will still try it 😂

**Comment by JokersCat** [comment]
Would you attempt it for money 🧐

**Comment by aw1875** [comment]
@JokersCat take a look [at this repo](https://github.com/h0nde/py-hcaptcha). I feel like this is what you’re looking for in Python.

**Comment by JokersCat** [comment]
Oh okay thx and do you got a tip for me how I could inject a hcaptcha token into a website? (with the browser console)

**Comment by aw1875** [comment]
Ya for sure, its actually super easy. I'll write an example using [pyppeteer](https://github.com/pyppeteer/pyppeteer) (a Python version that tries to replicate puppeteer). Once you get your token (however you decide to do that) you would do something like this:

```python
await page.evaluate(() => {
    document.querySelector('[name="h-captcha-response"]').value = "YOUR TOKEN";
});
```

I just want to note that you may have to look a little more into the documentation of pyppeteer to confirm this is accurate if you decide to use it over something else like selenium. But, regardless of what you choose the process should be relatively similar. You need to find the query selector with the name "h-captcha-response" on the page and set it's value to the returned token.

**Comment by JokersCat** [comment]
Oh okay thank you I used selenium before.There you can use a command something like driver.execute_script or something like that.

**Comment by JokersCat** [comment]
But that doesnt worked for me so idk :( I actually searched about this topic and found other people got the same issue but none of them got a solution :(

**Comment by aw1875** [comment]
As long as your token is right you just need to set the value of the queryselector I sent above. There is nothing else to do once you get the token

**Comment by aw1875** [comment]
I'm going to close this issue as it has been addressed 54c8451e9816318e49ce49369abba22fd70a85b0
